packages/material-ui/src/Select/SelectInput.js

```diff
@@ -15,17 +15,14 @@ import useForkRef from '../utils/useForkRef';
 import useControlled from '../utils/useControlled';
 import selectClasses, { getSelectUtilitiyClasses } from './selectClasses';
 
+// TODO: enable overrides for the icon and nativeInput slots via the components prop
 export const overridesResolver = (props, styles) => {
   const { styleProps } = props;
   return deepmerge(
     {
       ...styles.select,
```

Comment:
AFAIK, same issue with root and select, the Input wrapper CSS wins.

```suggestion
     [`&.${selectClasses.select}`]: styles.select,
```